compliance/node/index/indexer.go

```diff
@@ -109,12 +115,19 @@ type NodeIndexerConfig struct {
 }
 
 // DefaultNodeIndexerConfig is the default configuration for a node indexer.
-var DefaultNodeIndexerConfig = NodeIndexerConfig{
-	HostPath: env.NodeIndexHostPath.Setting(),
-	// The default, mTLS-capable client will be used.
-	Client:             nil,
-	Repo2CPEMappingURL: env.NodeIndexMappingURL.Setting(),
-	Timeout:            10 * time.Second,
+func DefaultNodeIndexerConfig() NodeIndexerConfig {
```

Comment:
Why is this a function now? All fields seem to be constants.

Reply:
This is a function for lazy evaluation and better testability. It does not strictly has to be a function - let me know if you want me to revert it back to a var.

Reply:
I have no preference. Please adjust the comment.

BTW, I think the naming convention for constructors is to start with `Make...`.

Reply:
I feel it sounds weird with `MakeDefaultNodeIndexerConfig()`. I would leave the name as it is - we use it like this also for functions.
![Screenshot 2025-04-01 at 11 55 24](https://github.com/user-attachments/assets/32ea68c8-b9b2-4963-8cfc-8c54f3794642)

Reply:
Here is the comment adjustment: [0063f48](https://github.com/stackrox/stackrox/pull/14766/commits/0063f48b402f5f190ae9bfcb47bb09f5a0bba771)